checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -648,24 +653,56 @@ private void addAnnotationForDivide(
             Integer maybeValRight = maybeValFromValueType(valueTypeRight);
             if (maybeValRight != null) {
                 AnnotatedTypeMirror leftType = getAnnotatedType(left);
-                addAnnotationForLiteralDivide(maybeValRight, leftType, type);
+                addAnnotationForLiteralDivide(maybeValRight, left, leftType, type);
             }
         }
 
         private void addAnnotationForLiteralDivide(
-                int val, AnnotatedTypeMirror nonLiteralType, AnnotatedTypeMirror type) {
-            if (nonLiteralType.hasAnnotation(LTLengthOf.class)
-                    || nonLiteralType.hasAnnotation(LTOMLengthOf.class)) {
+                int val,
+                ExpressionTree leftTree,
+                AnnotatedTypeMirror leftType,
+                AnnotatedTypeMirror type) {
+            if (isLTL(leftType)) {
                 if (val >= 1) {
-                    type.replaceAnnotation(nonLiteralType.getAnnotationInHierarchy(UNKNOWN));
+                    type.replaceAnnotation(leftType.getAnnotationInHierarchy(UNKNOWN));
                 }
-            } else if (nonLiteralType.hasAnnotation(LTEqLengthOf.class)) {
+            } else if (leftType.hasAnnotation(LTEqLengthOf.class)) {
                 // FIXME: Is this unsafe? What if the length is zero?
                 if (val >= 2) {
-                    String[] names = getValue(nonLiteralType.getAnnotationInHierarchy(UNKNOWN));
+                    String[] names = getValue(leftType.getAnnotationInHierarchy(UNKNOWN));
                     type.replaceAnnotation(createLTLengthOfAnnotation(names));
                 } else if (val == 1) {
-                    type.addAnnotation(nonLiteralType.getAnnotationInHierarchy(UNKNOWN));
+                    type.addAnnotation(leftType.getAnnotationInHierarchy(UNKNOWN));
+                }
+            } else if (val == 2) {
+                // The average of two LTL/LTOMs is LTL.
+
+                // Necessary because otherwise an expression in parentheses doesn't make it past
+                // the next check...
+                if (leftTree.getKind() == Tree.Kind.PARENTHESIZED) {
```

Comment:
I would make this a while loop, just in case. It's no more complicated but handles the case of an extra set of parentheses.  It's what is (or should be) done elsewhere.